assets/javascripts/initializers/code-bytes.js

```diff
@@ -175,8 +176,36 @@ function initializeCodeByte(api) {
         div.appendChild(saveButton);
       }
     });
-  }),
-    { id: 'codebyte-preview' };
+  },
+  { id: 'codebyte-preview' });
+
+  api.modifyClass("controller:composer", {
+    save(...args) {
+      let allCodebytesAreValid = true;
+      let index = 0;
+      let start, end;
+      const inputLines = this.model.reply.split('\n');
+
+      do {
+        [start, end] = findCodeByte(inputLines, index);
+        index++;
+        if (start !== undefined && !inputLines[start].match(/^\[codebyte[ ]+language=([^\s]*?)[ ]*]$/)) {
+          allCodebytesAreValid = false;
+        }
+      } while (allCodebytesAreValid && start !== undefined)
```

Comment:
I think this is the first `do... while` I've seen in production front end code lol (but it's the right tool!)